File: server/internal/handler/dev.go
package handler

import (
	"database/sql"
	"net/http"

	"cashcompass-server/internal/db"
)

type DevHandler struct {
	DB      *sql.DB
	DevMode bool
}

func NewDevHandler(database *sql.DB, devMode bool) *DevHandler {
	return &DevHandler{DB: database, DevMode: devMode}
}

func (h *DevHandler) RegisterRoutes(mux *http.ServeMux) {
	mux.HandleFunc("POST /dev/reset", h.handleReset)
}

func (h *DevHandler) handleReset(w http.ResponseWriter, r *http.Request) {
	if !h.DevMode {
		http.Error(w, "forbidden", http.StatusForbidden)
		return
	}
	for _, stmt := range []string{
		"DELETE FROM transactions",
		"DELETE FROM categories",
		"DELETE FROM accounts",
		"DELETE FROM sqlite_sequence WHERE name IN ('transactions','categories','accounts')",
	} {
		if _, err := h.DB.ExecContext(r.Context(), stmt); err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
	}
	if _, err := db.SeedIfEmpty(h.DB); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
	w.WriteHeader(http.StatusOK)
	_, _ = w.Write([]byte("ok"))
}

File: src/routes/dev.py
from __future__ import annotations
from fasthtml.common import *
from src.db import Database, seed_if_empty


def register(rt, db: Database, dev_mode: bool):
    if not dev_mode:
        return

    @rt("/dev/reset", methods=["POST"])
    def post():
        db.execute("DELETE FROM transactions")
        db.execute("DELETE FROM categories")
        db.execute("DELETE FROM accounts")
        db.execute("DELETE FROM sqlite_sequence WHERE name IN ('transactions','categories','accounts')")
        db.commit()
        seed_if_empty(db)
        return "ok"
